torch/distributions/multinomial.py

```diff
@@ -73,7 +74,7 @@ def sample(self, sample_shape=torch.Size()):
         samples = samples.permute(*shifted_idx)
         counts = samples.new(self._extended_shape(sample_shape)).zero_()
         counts.scatter_add_(-1, samples, torch.ones_like(samples))
-        return counts.type_as(self.probs)
+        return counts.type_as(self._param)
```

Comment:
(not part of this PR, but) Does `.type_as()` get GPU placement right? Or should we instead be using `self._param.new(...)`?